### PR TITLE
Removed constants file and did some renaming

### PR DESCRIPTION
**Pull request by pwillsey**
With his refactor I was trying to remove the global constants file and also make things a bit easier to understand by renaming some of the classes that were confusingly named. I removed YMConstants.h partly because I think constants should be located where they're actually used and also because this approach of having user defined constants won't work for a Cocoapod. Anything that needs to be set like the client id or client secret is now a property on YMLoginClient, which the user needs to set. This is similar to how Crittercism and HockeyApp work. I renamed YMHTTPClient to YMAPIClient and I renamed YMLoginController to YMLoginClient. It was confusing that YMLoginController wasn't actually a UIViewController and YMHTTPClient is a holdover from AFNetworking 1.0.

### CODE REVIEW

File: Example Project/SampleApp/YMSampleHomeViewController.m [hunk 1]
```diff
@@ -29,7 +28,16 @@ - (void)dealloc
 // This is called by clicking the login button in the sample interface.
 - (IBAction)login:(id)sender
 {
-    [[YMLoginController sharedInstance] startLogin];
+    /* Add your client ID here */
+    [[YMLoginClient sharedInstance] setAppClientID:@"APP CLIENT ID"];
```
Comment (joelklabo) [inline Example Project/SampleApp/YMSampleHomeViewController.m]: Maybe we should set these in the app delegate? We would just call startLogin here. I feel like that's the standard place I see this kind of thing.

**Comment by joelklabo** [comment]
I like it. Looks way better.

File: Example Project/SampleApp/YMSampleHomeViewController.m [hunk 2]
```diff
@@ -29,7 +28,16 @@ - (void)dealloc
...
+    
+    /* Add your client secret here */
+    [[YMLoginClient sharedInstance] setAppClientSecret:@"APP CLIENT SECRET"];
+    
+    /* Add your authorization redirect URI here */
+    [[YMLoginClient sharedInstance] setAuthRedirectURI:@"AUTH REDIRECT URI"];
+    
+    [[YMLoginClient sharedInstance] startLogin];
```
Comment (joelklabo) [inline Example Project/SampleApp/YMSampleHomeViewController.m]: Maybe we should set these in the app delegate? We would just call startLogin here. I feel like that's the standard place I see this kind of thing.

Comment (pwillsey) [inline Example Project/SampleApp/YMSampleHomeViewController.m]: Yeah I agree I'll add it to the app delegate

**Comment by dtweston** [comment]
:+1:

**Pull request merged**